### PR TITLE
gst_all_1.gst-plugins-good: fix building rpicamsrc

### PR DESCRIPTION
**Pull request by lopsided98**
###### Description of changes

An assertion was preventing `raspiCameraSupport` from being enabled on 32-bit ARM, and the build was failing because it needs to be told where to find `libraspberrypi`. 

Later, I also realized that `rpicamsrc` is actually not supported on aarch64, because the MMAL libraries in `libraspberrypi` don't support aarch64. I believe there was a period of time where MMAL could be built on aarch64, which led to this code in nixpkgs, but this support was removed upstream because it was buggy (see https://github.com/raspberrypi/userland/issues/688).

I tested that `rpicamsrc` works on a Raspberry Pi Zero.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [x] armv6l-linux (cross)
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of rpicamsrc
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @vikanezrimaya @nh2

### CODE REVIEW

**Review by vikanezrimaya — APPROVED** [review]

**Comment by lopsided98** [comment]
I fixed the conflicts.

**Review by lilyinstarlight — APPROVED** [review]
Looks good to me, thank you for this!

File: pkgs/development/libraries/gstreamer/good/default.nix [hunk 1]
```diff
@@ -45,7 +45,7 @@
 , enableDocumentation ? stdenv.hostPlatform == stdenv.buildPlatform, hotdoc
 }:
 
-assert raspiCameraSupport -> (stdenv.isLinux && stdenv.isAarch64);
+assert raspiCameraSupport -> (stdenv.isLinux && stdenv.isAarch32);
```
Comment (lilyinstarlight) [inline pkgs/development/libraries/gstreamer/good/default.nix:48]: Actually this is a bit of a minor nit, but do you think you add a comment to this linking to the MMAL issue?

That way it's clear why it won't work on aarch64 even when the upstream gstreamer meson file does imply it is supported on aarch32 and aarch64

**Review by lilyinstarlight — APPROVED** [review]
Awesome, thank you again!

**Pull request merged**